text/0019-milestone-payload/0019-milestone-payload.md

```diff
@@ -0,0 +1,58 @@
++ Feature name: `Milestone Payload`
++ Start date: 2020-07-28
++ RFC PR: [iotaledger/protocol-rfcs#0019](https://github.com/iotaledger/protocol-rfcs/pull/19)
++ Author: Angelo Capossele
+
+# Summary
+
+In the IOTA protocol, nodes use the milestones issued by the Coordinator to reach a consensus on which transactions are confirmed. 
+This RFC proposes a new milestone payload as well as the EdDSA signature algorithm as a replacement to the current Coordinator digital signature scheme to authenticate the issued milestones.
+
+# Motivation
+The current signature scheme used to authenticate milestones issued by the Coordinator has been designed with ternary and quantum robustness in mind. Due to the switch to binary, the new message layout as well as the upcoming Coordicide (i.e., Coordinator removal), we now have the opportunity to revisit such a mechanism and replace it with a more simple, well-vetted and standardized one. Although the proposed digital signature scheme, EdDSA, does not provide quantum robustness, its simplicity and easier key-managment make it a good candidate for the time being.
+
+# Detailed design
+
+The EdDSA signature algorithm in its variants **Ed25519** and **Ed448** (providing a security level of 128-bit and 224-bit respectively) are technically described in the IRTF [RFC 8032](https://tools.ietf.org/html/rfc8032).
+Size of both private and public keys are of 32 or 57 bytes depending on the curve used. Similarly, the signature size is of 64 or 114 bytes.
+
+- The Ed25519 key generation can be fed with a given random sequence of bytes (i.e., seed) of length 32 bytes. Output of the key generation is the pair of private and public keys. Note that the generation of the seed is out of the scope of this RFC, but in general, any cryptographic pseudo-random function (PRF) would suffice.
+- The Ed25519 signature function takes a Ed25519 private key, a sequence of bytes of arbitrary size and produces an Ed25519 signature of length 64 bytes. The given sequence of bytes should then be internally hashed (using sha512) by the same function.
+- The Ed25519 verification function takes a public key, a sequence of bytes of arbitrary size, a Ed25519 signature, and returns true/false based on the signature validity.
+
+To generate a valid milestone, the Coordinator *MUST*: 
+1. Generate a *Message* as defined in [RFC-0017 (draft)](https://github.com/GalRogozinski/protocol-rfcs/blob/message/text/0017-message/0017-message.md).
+2. Generate a new [milestone payload](#Milestone-payload) without filling the signature field.
+3. Sign the serialized bytes given by the concatenation of the following fields:
+    - Version, Parent1, Parent2, Payload Length of the Message;
```

Comment:
I understand why you do this, but it actually breaks the "layer" concept of the messages.

Reply:
I agree, this is certainly not perfect. However, I'd also say that it is not that dramatic as this references only a layer down (and not up which would indeed be a complete paradigm shift).

For a milestone, we need to make 100% sure that the branch/trunk (and probably also the message type) cannot be forged. 

@Alex6323 do you have any alternative idea to achieve this?
Copying those fields into the MS payload also does not seem very good.

Reply:
There is no a cleaner separation of layers, as the Milestone Payload contains the reference parent1 and parent2.